File: backend/app/schemas/fitness_score.py
from datetime import date, datetime
from typing import List, Optional
from uuid import UUID
from pydantic import BaseModel, ConfigDict, Field


class FitnessScoreItem(BaseModel):
    id: UUID
    user_id: UUID
    score: int = Field(..., ge=0, le=100)
    workout_adherence_pct: Optional[float] = None
    nutrition_score: Optional[float] = None
    protein_score: Optional[float] = None
    sleep_score: float = 75.0
    recovery_score: float = 75.0
    consistency_score: Optional[float] = None
    calculated_at: datetime
    period_start: date
    period_end: date

    model_config = ConfigDict(from_attributes=True)


class FitnessScoreResponse(BaseModel):
    current_score: Optional[FitnessScoreItem] = None
    score_label: str = "Needs Work"  # 'Excellent', 'Good', 'Fair', 'Needs Work'
    history: List[FitnessScoreItem] = []

File: src/types/fitnessScore.ts
export interface FitnessScoreItem {
  id: string;
  user_id: string;
  score: number;
  workout_adherence_pct?: number | null;
  nutrition_score?: number | null;
  protein_score?: number | null;
  sleep_score: number;
  recovery_score: number;
  consistency_score?: number | null;
  calculated_at: string;
  period_start: string;
  period_end: string;
}

export interface FitnessScoreResponse {
  current_score?: FitnessScoreItem | null;
  score_label: 'Excellent' | 'Good' | 'Fair' | 'Needs Work';
  history: FitnessScoreItem[];
}
